lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb

```diff
@@ -164,14 +164,18 @@ def self.expose(*args)
         self.association = method_name if options[:association]
         define_method(method_name) do |*params|
           method = options[:method] || method_name
-          ret = object_send(method, *params)
+          ret = User.with_user(self.class.workspace.ae_user) { object_send(method, *params) }
```

Comment:
From inside a class method you want just `workspace.ae_user`.  The `self.class` part would reference the class of the `MiqAeMethodService::MiqAeServiceModelBase` class which is `Class` and would not respond to the `workspace` method.

Reply:
Seems the exposed method runs as an instance method. Without ```self.class``` it would run into an error:
```
[----] W, [2019-03-13T14:04:55.710817 #12084:f06f54]  WARN -- : Error during substitution: undefined local variable or method `workspace' for #<MiqAeMethodService::MiqAeServiceMiqEvent:0x0000000008449b40>
```